### ISSUE TITLE
Ecosystem tour: Common crates to use with Tokio

### ISSUE THREAD

**Issue by carllerche** [tokio-rs/doc-blitz#34]
Building an application with Tokio usually means pulling in additional libraries. This guide section will provide an introduction to common crates and show how to use them from the context of an app built with Tokio.

Crates to include:

* [Hyper](https://github.com/hyperium/hyper)
* [Trust DNS](https://github.com/bluejekyll/trust-dns)

What else?

**Comment by kadfak** [comment]
Maybe there should also be an overview of they many tokio subcrates in addition to third party libraries. Right now one can only find that information -- that the project consists of many smaller components -- from tokio's [Github repository](https://github.com/tokio-rs/tokio#project-layout). This would open up the philosophy behind tokio a little bit more and form the idea that tokio is the symbiosis of many smaller crates. It might also help motivate library authors to use these individual crates when they know such architecture exists from the get-go and not the single tokio crate as brought up in #42.

**Comment by carllerche** [comment]
@kadfak I agree, my current thought was to discuss in this in the opening of the "internals" section [here](https://github.com/tokio-rs/doc-push/blob/master/outline/internals.md#overview). The "Tokio for LIbraries" section could reference this. Perhaps that is not the best place. Thoughts?
